field/composite.go

```diff
@@ -347,6 +368,52 @@ func (f *Composite) UnmarshalJSON(b []byte) error {
 }
 
 func (f *Composite) pack() ([]byte, error) {
+	if f.Bitmap() != nil {
+		return f.packByBitmap()
+	}
+
+	return f.packByTag()
+}
+
+func (f *Composite) packByBitmap() ([]byte, error) {
+	packed := []byte{}
+	f.Bitmap().Reset()
+
+	ids, err := f.packableFieldIDs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to pack composite: %w", err)
+	}
+
+	for _, id := range ids {
```

Comment:
// Set bitmap bits for all fields that are present.
for _, id := range ids {

Reply:
Listo